src/main/java/com/twitter/kinesis/utils/Environment.java

```diff
@@ -145,7 +161,25 @@ private double getDoubleProperty(String propName, double def) {
     return def;
   }
 
+  private Boolean getBooleanProperty(String propName, Boolean def) {
+    String prop = getStringProperty(propName);
+    if (prop != null) {
+      return Boolean.parseBoolean(prop);
+    }
+
+    return def;
+  }
+
   private static String propNameToEnvVar(String propName) {
     return propName.toUpperCase().replace('.', '_');
   }
+
+  private Date getReplayDate(String gnipDateString) {
+    try {
+      return DATE_FORMAT.parse(getStringProperty(gnipDateString));
+    } catch (ParseException e) {
```

Comment:
I'm thinking we don't catch this exception, if the date's improperly formatted it shouldn't have the same behavior as not having supplied it at all.